src/land/UberArcanistSubmitQueueEngine.php

```diff
@@ -25,15 +26,20 @@ public function execute() {
       $this->updateRevision();
 
       $this->pushChange();
-      $this->reconcileLocalState();
-
-      if ($this->getShouldKeep()) {
-        echo tsprintf(
-        "%s\n",
-        pht('Keeping local branch.'));
+      if ($this->shouldShadow) {
```

Comment:
Avoid changing existing code to make code easier to merge with upstream. Instead write like this:
```
if ($this->shouldShadow) {
  return;
}
```